tests/test_transpiler_optimizer.py

```diff
@@ -43,3 +49,352 @@ def test_fusion():
     fusion = Rearrange(max_qubits=1)
     fused_circ = fusion(circuit)
     assert isinstance(fused_circ.queue[0], gates.Unitary)
+
+
+def test_inversecancellation_no_gates():
+    circ = Circuit(1)
+    inverse_cancellation = InverseCancellation()
+    new_circuit = inverse_cancellation(circuit=circ)
+    assert new_circuit.ngates == 0
+    assert new_circuit.nqubits == 1
+
+
+def test_inversecancellation_gatesX():
+    circ = Circuit(1)
+    circ.add(gates.X(0))
+    circ.add(gates.X(0))
+    inverse_cancellation = InverseCancellation()
+    new_circuit = inverse_cancellation(circuit=circ)
+    assert new_circuit.ngates == 0
+    assert new_circuit.nqubits == 1
+
+
+def test_inversecancellation_gatesY():
+    circ = Circuit(1)
+    circ.add(gates.Y(0))
+    circ.add(gates.Y(0))
+    inverse_cancellation = InverseCancellation()
+    new_circuit = inverse_cancellation(circuit=circ)
+    assert new_circuit.ngates == 0
+    assert new_circuit.nqubits == 1
+
+
+def test_inversecancellation_gatesZ():
+    circ = Circuit(1)
+    circ.add(gates.Z(0))
+    circ.add(gates.Z(0))
+    inverse_cancellation = InverseCancellation()
+    new_circuit = inverse_cancellation(circuit=circ)
+    assert new_circuit.ngates == 0
+    assert new_circuit.nqubits == 1
+
+
+def test_inversecancellation_gatesH():
+    circ = Circuit(1)
+    circ.add(gates.H(0))
+    circ.add(gates.H(0))
+    inverse_cancellation = InverseCancellation()
+    new_circuit = inverse_cancellation(circuit=circ)
+    assert new_circuit.ngates == 0
+    assert new_circuit.nqubits == 1
+
+
+def test_inversecancellation_gatesCNOT():
+    circ = Circuit(2)
+    circ.add(gates.CNOT(0, 1))
+    circ.add(gates.CNOT(0, 1))
+    inverse_cancellation = InverseCancellation()
+    new_circuit = inverse_cancellation(circuit=circ)
+    assert new_circuit.ngates == 0
+    assert new_circuit.nqubits == 2
```

Comment:
Checking the number of gates in the new circuit is not a resilient enough test. A better test, for instance, would be to check if both circuits are implementing the same unitary, e.g.

```suggestion
def test_inversecancellation_gatesCNOT(backend):
    circ = Circuit(2)
    circ.add(gates.CNOT(0, 1))
    circ.add(gates.CNOT(0, 1))
    target = circ.unitary(backend)
    
    inverse_cancellation = InverseCancellation()
    new_circuit = inverse_cancellation(circuit=circ)
    unitary = new_circuit.unitary(backend)
    
    backend.assert_allclose(unitary, target)
```
The same comment is valid for all the other tests.

Reply:
I think the idea is great. I updated all tests, added new ones, and renamed test functions to conform to PEP 8 snake_case style.